### ISSUE TITLE
Linking locationIDs to existing

### ISSUE THREAD

**Issue by anders-kolstad** [NTNU-VM/natronR#5]
Flat data typically only have verbatimLocality (common names) not locationID (data names, long number), 
and it's too much work to match UUIDs from the Natron locations table on the flat data. Its probably better to match/compare verbatimLocality in the eventUpsert function, and abort or give a warning if the verbatimLocations doesn't already exist in the database. Then it's either a typo or the location is a new one, in which case we must add it with a localityUpsert function (not made yet) where we simulatneously assign new UUIDs. Any other ideas?

**Comment by andersfi** [comment]
Good idea that also holds generic across other ID columns (eg. sampleProtocollID). However, this means that the location are registred in the database up-front. If not, one have to add the location first. If adding location then the UUID for the location (locationID) should be set in the original data-set and one first have to upload the locations etc... 

### Technical implications: 
1. For natronbatchupload: Probably best to keep these functions separated in own compartment (mapping-part). 
2. For datamanagment/natron structure in general: There should be identified a field that should be used for the human readable locationID. This must then have a unique constraint across the location table. verbatimLocation is a candidate field (I much prefer to recirculate DwC terms than to invent new ones if possible). 

We are likely to run into several other issues in time when starting to using the NaTron database for real. @vmzomdav: My suggestion is that we start a new repository on the githubsite github.com/NTNU-VM called NaTron and start issue tracking of db specific generic changes there.

**Comment by vmzomdav** [comment]
It should  not be necessary to have a “human readable locationID”, the uuid “locationID” is already well documented by the content of the other columns, the xy coordinate in association with locality, verbatimLocality, stationNumber and siteNumber as well as the other columns are already a secondary key/id to the locality geographical point, I am afraid that generating another key as serial human number starting from 1, 2, 3… to keep track of locality names will overload register with unreadable number when having several thousands localities. .

**Comment by andersfi** [comment]
My (and I believe @anders-kolstad's point) is that when punching and editing data manually that have pre-defined locationIDs it is convenient to have a more human readable column to relate to. This is equivalent to using vatn_lnr instead of locationID in NOFA when entering data there, (or feks when recording samplingProtocol, where you don't as a human should interfere to much with the UUID). It does not mather if this is the siteID, stationNumber, verbatimLocation or others that can be used in original data recording and then resolved against the locationID when mapping data to the database. However, the procedure for doing this should be clear, including what fields that can be used in this way.

**Comment by samaperrin** [comment]
I'm with Marc, I don't think generating another id will help. It should be fairly easy to write 4-5 lines (max) of code that matches whatever 'human-readable' id we have for the location and match it to a uuid. Just use a dployr/dbplyr function to filter the already existing locations based on whatever human-readable id we have and get the corresponding uuid as output.

**Comment by anders-kolstad** [comment]
Then I think we all agree. As long as the locations already exist in Natron, the datasets we upload don't need UUID's (it's very quick to make mistakes when manually handeling these IDs). Insted we do what Sam says above. Candiate fields are 'verbatimLocation' as Finstad said, or just 'Location'.

**Comment by andersfi** [comment]
Yepp. Just make sure that you know (i.e. also in time document this some place) which field can be used for this purpose and that those fields, or combination of fields, have a unique constraint in the db (or else we may end up in misery)
